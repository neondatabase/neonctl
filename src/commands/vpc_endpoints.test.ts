import { describe } from 'vitest';
import { test } from '../test_utils/fixtures';

describe('vpc-endpoints', () => {
  test('list org VPC endpoints', async ({ testCliCommand }) => {
    await testCliCommand(
      ['vpc', 'endpoint', 'list', '--org-id', '1', '--region-id', 'test'],
      { mockDir: 'single_org' },
    );
  });

  test('list org VPC endpoints implicit org', async ({ testCliCommand }) => {
    await testCliCommand(['vpc', 'endpoint', 'list', '--region-id', 'test'], {
      mockDir: 'single_org',
    });
  });

  test('update org VPC endpoint', async ({ testCliCommand }) => {
    await testCliCommand(
      [
        'vpc',
        'endpoint',
        'update',
        'vpc-test',
        '--label',
        'newlabel',
        '--region-id',
        'test',
      ],
      { mockDir: 'single_org' },
    );
  });

  test('delete org VPC endpoint', async ({ testCliCommand }) => {
    await testCliCommand(
      ['vpc', 'endpoint', 'remove', 'vpc-test', '--region-id', 'test'],
      { mockDir: 'single_org' },
    );
  });

  test('get org VPC endpoint status', async ({ testCliCommand }) => {
    await testCliCommand(
      ['vpc', 'endpoint', 'status', 'vpc-test', '--region-id', 'test'],
      { mockDir: 'single_org' },
    );
  });

  test('set org VPC endpoint in azure', async ({ testCliCommand }) => {
    await testCliCommand(
      [
        'vpc',
        'endpoint',
        'update',
        'vpc-test',
        '--label',
        'newlabel',
        '--region-id',
        'azure-test',
      ],
      {
        mockDir: 'single_org',
        stderr:
          'INFO: VPC endpoint configuration is not supported for Azure regions',
      },
    );
  });

  test('list project VPC endpoint restrictions', async ({ testCliCommand }) => {
    await testCliCommand(
      [
        'vpc',
        'project',
        'list',
        '--region-id',
        'test',
        '--project-id',
        'test-project-123456',
      ],
      { mockDir: 'single_org' },
    );
  });

  test('list project VPC endpoint restrictions with implicit project', async ({
    testCliCommand,
  }) => {
    await testCliCommand(['vpc', 'project', 'list', '--region-id', 'test'], {
      mockDir: 'single_org',
    });
  });

  test('set project VPC endpoint restrictions', async ({ testCliCommand }) => {
    await testCliCommand(
      ['vpc', 'project', 'update', 'vpc-test', '--label', 'newlabel'],
      { mockDir: 'single_org' },
    );
  });

  test('delete project VPC endpoint restrictions', async ({
    testCliCommand,
  }) => {
    await testCliCommand(['vpc', 'project', 'remove', 'vpc-test'], {
      mockDir: 'single_org',
    });
  });
});
