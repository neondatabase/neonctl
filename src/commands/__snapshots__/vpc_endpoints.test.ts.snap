// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`vpc-endpoints > delete org VPC endpoint 1`] = `
"{}
"
`;

exports[`vpc-endpoints > delete project VPC endpoint restrictions 1`] = `
"{}
"
`;

exports[`vpc-endpoints > get org VPC endpoint status 1`] = `
"vpc_endpoint_id: vpc-test
state: new
"
`;

exports[`vpc-endpoints > list org VPC endpoints 1`] = `
"- vpc_endpoint_id: vpc-test
  label: test-label
"
`;

exports[`vpc-endpoints > list org VPC endpoints implicit org 1`] = `
"- vpc_endpoint_id: vpc-test
  label: test-label
"
`;

exports[`vpc-endpoints > list project VPC endpoint restrictions 1`] = `
"- vpc_endpoint_id: vpc-test
  label: test-label-project
"
`;

exports[`vpc-endpoints > list project VPC endpoint restrictions with implicit project 1`] = `
"- vpc_endpoint_id: vpc-test
  label: test-label-project
"
`;

exports[`vpc-endpoints > set org VPC endpoint in azure 1`] = `
"{}
"
`;

exports[`vpc-endpoints > set project VPC endpoint restrictions 1`] = `
"{}
"
`;

exports[`vpc-endpoints > update org VPC endpoint 1`] = `
"{}
"
`;
